src/Lachain.Storage/Repositories/LocalTransactionRepository.cs

```diff
@@ -31,11 +35,55 @@ public void SaveState(byte[] state)
 
         public void TryAddTransaction(TransactionReceipt receipt)
         {
-            if (_watchAddresses.Count(addr =>
-                    addr.Equals(receipt.Transaction.To) || addr.Equals(receipt.Transaction.From)) <= 0) return;
-            
-            var data = LoadState();
-            SaveState(data.Concat(receipt.Hash.ToBytes()).ToArray());
+            string[] signatures =
+            {
+                "transfer(address,uint256)",
+                "transferFrom(address,address,uint256)",
+                "mint(address,uint256)",
+                "burn(address,uint256)"
+            };
+
+            var decoder = new ContractDecoderLtr(receipt.Transaction.Invocation.ToArray());
+            object[] decodedRes = Array.Empty<object>();
+
+            foreach (var signature in signatures)
+            {
+                try
+                {
+                    decodedRes = decoder.Decode(signature);
+
+                    if (decodedRes.Length == 3)
+                    {
+                        if (_watchAddresses.Count(addr =>
+                            decodedRes[0] is UInt160 fromAddr && fromAddr.Equals(addr) ||
+                            decodedRes[1] is UInt160 toAddr && toAddr.Equals(addr)) <= 0) continue;
+
+                        var temp = LoadState();
+                        SaveState(temp.Concat(receipt.Hash.ToBytes()).ToArray());
+                    }
+
+                    else if (decodedRes.Length == 2)
+                    {
+                        if (_watchAddresses.Count(addr =>
+                            decodedRes[0] is UInt160 fromAddr && fromAddr.Equals(addr)) <= 0) continue;
+
+                        var temp = LoadState();
+                        SaveState(temp.Concat(receipt.Hash.ToBytes()).ToArray());
+                    }
+                    else
+                    {
+                        if (_watchAddresses.Count(addr =>
```

Comment:
This code will be executed really rarely. For the most part of transactions (for example,  transactions without contract calls) we get exception during decoder.Decode(signature) call and goes to the next signature without checking this condition. This code should be outside the try-catch block (and after the cycle really).